cmd/osmosisd/cmd/root.go

```diff
@@ -477,6 +482,77 @@ func overwriteConfigTomlValues(serverCtx *server.Context) error {
 	return nil
 }
 
+// overwrites app.toml values. Returns error if app.toml does not exist
+//
+// Currently, overwrites arbitrage-min-gas-fee value in app.toml if it is set to 0.005. Similarly,
+// overwrites the given viper config value.
+// Silently handles and skips any error/panic due to write permission issues.
+// No-op otherwise.
+func overwriteAppTomlValues(serverCtx *server.Context) error {
+	// Get paths to config.toml and config parent directory
+	rootDir := serverCtx.Viper.GetString(tmcli.HomeFlag)
+
+	configParentDirPath := filepath.Join(rootDir, "config")
+	configFilePath := filepath.Join(configParentDirPath, "app.toml")
+
+	fileInfo, err := os.Stat(configFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read in %s: %w", configFilePath, err)
+	} else {
+		// app.toml exists
+
+		// Get setting
+		currentArbitrageMinGasFeeValue := serverCtx.Viper.Get(mempoolConfigName + "." + arbitrageMinGasFeeConfigName)
+
+		// .x format at 0.x format are both valid.
+		if currentArbitrageMinGasFeeValue == oldArbitrageMinGasFeeValue || currentArbitrageMinGasFeeValue == "0"+oldArbitrageMinGasFeeValue {
+			// Set new value in viper
+			serverCtx.Viper.Set(mempoolConfigName+"."+arbitrageMinGasFeeConfigName, newArbitrageMinGasFeeValue)
+
+			defer func() {
+				if err := recover(); err != nil {
+					fmt.Printf("failed to write to %s: %s\n", configFilePath, err)
+				}
+			}()
+
+			// Check if the file is writable
+			if fileInfo.Mode()&os.FileMode(0200) != 0 {
+				// Read the entire content of the file
+				content, err := os.ReadFile(configFilePath)
+				if err != nil {
+					return err
+				}
+
+				// Convert the content to a string
+				fileContent := string(content)
+
+				// Find the index of the search line
+				index := strings.Index(fileContent, arbitrageMinGasFeeConfigName)
+				if index == -1 {
+					return fmt.Errorf("search line not found in the file")
+				}
+
+				// Find the opening and closing quotes
+				openQuoteIndex := strings.Index(fileContent[index:], "\"")
+				openQuoteIndex += index
+
+				closingQuoteIndex := strings.Index(fileContent[openQuoteIndex+1:], "\"")
+				closingQuoteIndex += openQuoteIndex + 1
+
+				// Replace the old value with the new value
+				newFileContent := fileContent[:openQuoteIndex+1] + newArbitrageMinGasFeeValue + fileContent[closingQuoteIndex:]
```

Comment:
Note:

The following are the reasons why this is done manually:
- https://github.com/pelletier/go-toml does not handle comments when writing. So all config comments would be removed if I were to load the file, update the value, and overwrite.
- SDK helpers either unexpectedly overwrite other config values or run into the same comment issue as above. I found it easier to simply string-match

Reply:
https://github.com/cosmos/cosmos-sdk/issues/19257

Reply:
I also ran into this when I attempted this a while ago, thanks for making an issue